lib/apartmentex.ex

```diff
@@ -1,24 +1,18 @@
 defmodule Apartmentex do
-
   @schema_prefix Application.get_env(:apartmentex, :schema_prefix) || "tenant_"
   @tenant_migration_folder "priv/repo/tenant_migrations"
   #warning: don't change this after you already have tenants
 
-
-  alias Ecto.Repo.Queryable
-  alias Ecto.Repo.Model
   alias Ecto.Changeset
-  alias Apartmentex.Helpers.Queryable, as: QHelpers
-  alias Apartmentex.Helpers.Model, as: MHelpers
-  require Ecto.Query
 
   def new_tenant(repo, tenant) do
     prefix = build_prefix(tenant)
     case repo.__adapter__ do
       Ecto.Adapters.Postgres -> Ecto.Adapters.SQL.query(repo, "CREATE SCHEMA #{prefix}", [])
       Ecto.Adapters.MySQL -> Ecto.Adapters.SQL.query(repo, "CREATE DATABASE #{prefix}", [])
     end
-    Apartmentex.Migrator.run(repo, @tenant_migration_folder, :up, [all: true, prefix: String.to_atom(prefix)])
+    Ecto.Migration.Supervisor.start_link
+    Ecto.Migrator.run(repo, @tenant_migration_folder, :up, [all: true, prefix: String.to_atom(prefix)])
```

Comment:
The Ecto Migration Supervisor should run with the ecto app and will be supervised by whatever mix project is supervising ecto:

https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/application.ex

Should have no problem assuming it is running since none of the other apartmentex functions will work without ecto either.  I think it should be fine to explicitly start it just for the tests. 

I would do this before the tests to ensure ecto is up and running (and by extension the migration supervisor): Application.ensure_all_started(:ecto).  we could put that in normal apartmentex code as well, but I can't think of a situation where ecto wouldn't be in the supervision tree of the main application.   non-phoenix projects maybe?